### ISSUE TITLE
I dont know what is wrong...

### ISSUE THREAD

**Issue by Bhongcheng** [JedMills/MTFL-For-Personalised-DNNs#7]
I follow the setting in your .md file, but I find something very strange.

I try to run 'dset-cifar10_alg-fedavg-adam_C-0.5_B-20_T-200_E-1_device-gpu_W-200_seed-0_lr-0.1_noisy_frac-0.0_bn_private-yb'
the result of acc for train and test is about 45% and 10%

for 'dset-cifar10_alg-fedavg_C-0.5_B-20_T-200_E-1_device-gpu_W-200_seed-0_lr-0.1_noisy_frac-0.0_bn_private-none'
the result of acc for train and test is about 93% and 10%

I don't remember I have changed any code... What I do is only set the parameters in the main procedure instead of command-line. 

Cloud you please give me some suggestion?

**Comment by JedMills** [comment]
Hi Bhongcheng,

The first command that you quote is using the MTFL(FedAvg-Adam) optimiser, whereas the second is using MTFL(FedAvg). If you read the caption of Fig. 5 of the paper (for which the "fig-5-settings.md" file contains the hyperparameters), those results are using MTFL(FedAvg), not MTFL(FedAvg-Adam).  

Looking at Fig. 5, the results are around 93% accuracy, which is what you said that you achieved. I expect that the learning rate you used for MTFL(FedAvg-Adam) is too high: the default learning rate for the Adam optimiser is 0.001. 

I'm not sure about your problem with the train vs. test accuracy. Do you have any further details?

Warm regards,
Jed

**Comment by Bhongcheng** [comment]
I really appreciate it for your reply!

I read the result file(.pkl)
![image](https://user-images.githubusercontent.com/90205287/185923317-bcd65692-2faf-4994-9591-7f130a52b516.png)
"dset-cifar10_alg-fedavg_C-0.5_B-20_T-200_E-1_device-gpu_W-200_seed-0_lr-0.1_noisy_frac-0.0_bn_private-none.pkl"


This is the comment of the algorithm
![image](https://user-images.githubusercontent.com/90205287/185923732-92539680-a911-4365-8cd7-96cfe108d258.png)


here is part of the result
![image](https://user-images.githubusercontent.com/90205287/185923061-57aafdec-8ff4-4584-a1b9-5fd282460843.png)
![image](https://user-images.githubusercontent.com/90205287/185923101-b90578c8-fd2f-41e4-b76b-3fa5a1871b29.png)
![image](https://user-images.githubusercontent.com/90205287/185923154-07cb954a-30d9-4ebd-a37b-294b8cc8a7e7.png)
![image](https://user-images.githubusercontent.com/90205287/185923196-af915fd0-5370-4adf-9e5a-94eb9a7e735d.png)
So I think the result is '(train_accs is about 0.93 and test_accs is about 10%)'

If I get something wrong, could you please help me?

Thank you!

**Comment by JedMills** [comment]
Hi Bhongchen, 

Thanks for the further info. It looks like the model is training correctly (train err decreasing, train acc increasing), but the test result is just random. How are you loading in the train and test data? 

Jed

**Comment by Bhongcheng** [comment]
![image](https://user-images.githubusercontent.com/90205287/185944331-33a8a61b-d210-45f6-93d7-d76c08a5f74b.png)
just follow by your code

**Comment by JedMills** [comment]
Hi Bhongcheng,

I've just fixed a bug in `data_utils.py` for loading the CIFAR10 test set, which I had made when altering the code for GitHub publication. After the fix I ran the code with the following settings:

`python .\main.py -dset cifar10 -alg fedavg -C 0.5 -B 20 -T 200 -E 1 -W 200 -seed 0 -lr 0.1 -bn_private none -noisy_frac 0.0 -device gpu`,

and the test accuracy increased to the expected values. Please could you try running again with the fix to see if it works now?

Warm regards,
Jed

**Comment by Bhongcheng** [comment]
It works, thank you for your warm help!

**Comment by JedMills** [comment]
That's no problem, thank you for highlighting the issue.
